### PR TITLE
sf_i6_2 decompiled

### PR DESCRIPTION
**Pull request by inspectredc**

### CODE REVIEW

File: src/overlays/ovl_i6/sf_i6_2.c [hunk 1]
```diff
@@ -1,17 +1,474 @@
...
+            if (D_8015F908 < 600) {
+                player->camEye.x += actor4->vel.x * 0.23f;
+                player->camEye.y += actor4->vel.y * 0.23f;
+                player->camEye.z += actor4->vel.z * 0.23f;
+            }
+            Math_SmoothStepToF(&player->camAt.x, actor4->obj.pos.x, 1.0f, 20000.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.y, actor4->obj.pos.y, 1.0f, 20000.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.z, actor4->obj.pos.z, 1.0f, 20000.0f, 0.0f);
+            Math_SmoothStepToF(&player->unk_034, 0, 0.1f, 0.2f, 0.0f);
+            if ((gControllerPress->button & 0x1000) || (D_8015F928 == (D_800C9B4C + 300))) {
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/sf_i6_2.c:136]:
```suggestion
if ((gControllerPress->button & START_BUTTON) || (D_8015F928 == (D_800C9B4C + 300))) {

**Review by sonicdcer — APPROVED** [review]
Beautiful, i'll leave you a tiny nit :)

File: src/overlays/ovl_i6/sf_i6_2.c (hunk 1 above)
Comment (inspectredc) [reply src/overlays/ovl_i6/sf_i6_2.c:136]: omg i saw that on the scratch but forgot to check if we had a macro 🤦🏼

File: src/overlays/ovl_i6/sf_i6_2.c [hunk 2]
```diff
@@ -1,17 +1,474 @@
...
+                if (D_80178340 == 0xFF) {
+
+                    for (i = 1; i < ARRAY_COUNT(gTeamShields); i++) {
+                        D_80177C38[i] = gSavedTeamShields[i];
+                        D_801778F0[i] = gSavedTeamShields[i];
+                        gSavedTeamShields[i] = gTeamShields[i];
+                    }
+
+                    gNextGameState = 7;
+                    gNextLevel = 9;
```
Comment (petrie911) [inline src/overlays/ovl_i6/sf_i6_2.c:415]:
```suggestion
                    gNextGameState = GSTATE_PLAY;
                    gNextLevel = LEVEL_VENOM_ANDROSS;
```

File: src/overlays/ovl_i6/sf_i6_2.c [hunk 3]
```diff
@@ -1,17 +1,474 @@
...
+            if (gCsFrameCount == 105) {
+                Audio_PlaySfx(0x09000002, player->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                player->unk_194 = 5.0f;
+                player->unk_190 = 5.0f;
+            }
+            if (gCsFrameCount > 110) {
+                D_80178358 = 255;
+                D_8017835C = 8;
+                D_80178348 = D_80178350 = D_80178354 = 0;
+                if (D_80178340 == 0xFF) {
```
Comment (petrie911) [inline src/overlays/ovl_i6/sf_i6_2.c:406]:
```suggestion
                if (D_80178340 == 255) {
```

**Review by petrie911 — COMMENTED** [review]
just two small things I noticed

**Pull request merged**